source/dub/packagesuppliers/filesystem.d

```diff
@@ -59,15 +62,14 @@ class FileSystemPackageSupplier : PackageSupplier {
 		import dub.recipe.json : toJson;
 
 		auto filePath = bestPackageFile(packageId, dep, pre_release);
-		string packageFileName;
-		string packageFileContent = packageInfoFileFromZip(filePath, packageFileName);
-		auto recipe = parsePackageRecipe(packageFileContent, packageFileName);
-		Json json = toJson(recipe);
 		auto basename = filePath.head.name;
 		enforce(basename.endsWith(".zip"), "Malformed package filename: " ~ filePath.toNativeString);
 		enforce(basename.startsWith(packageId), "Malformed package filename: " ~ filePath.toNativeString);
-		json["version"] = basename[packageId.length + 1 .. $-4];
-		return json;
+
+		string packageFileName;
+		string packageFileContent = packageInfoFileFromZip(filePath, packageFileName);
+		return typeof(return)(parsePackageRecipe(packageFileContent, packageFileName));
+		// recipe.version_ = basename[packageId.length + 1 .. $-4];
```

Comment:
Setting the `version_` is important, why was it removed ?

Reply:
seems like we don't have any `version_` field inside PackageRecipe

Reply:
Last I checked we did. It's used when packages are downloaded from the registry and stored locally. That's how `dub` knows what version a package is at (it doesn't rely on the path).